core/server/src/main/java/alluxio/master/file/FileSystemMaster.java

```diff
@@ -1329,18 +1334,17 @@ private void deleteRecursiveInternal(LockedInodePath inodePath, boolean replayed
    * Implements file deletion.
    *
    * @param inodePath the file {@link LockedInodePath}
-   * @param recursive if the file id identifies a directory, this flag specifies whether the
-   *        directory content should be deleted recursively
+   * @param deleteFileOptions method optitions
    * @param replayed whether the operation is a result of replaying the journal
    * @param opTimeMs the time of the operation
    * @throws FileDoesNotExistException if a non-existent file is encountered
    * @throws IOException if an I/O error is encountered
    * @throws InvalidPathException if the specified path is the root
    * @throws DirectoryNotEmptyException if recursive is false and the file is a nonempty directory
    */
-  private void deleteInternal(LockedInodePath inodePath, boolean recursive, boolean replayed,
-      long opTimeMs) throws FileDoesNotExistException, IOException, DirectoryNotEmptyException,
-      InvalidPathException {
+  private void deleteInternal(LockedInodePath inodePath, DeleteFileOptions deleteFileOptions,
```

Comment:
options should be the last argument and operation time should be folded into the options object